docs/resources/aws_security_group.md.erb

```diff
@@ -224,55 +234,78 @@ The `allow` series of matchers enable you to perform queries about what network
-  * to_port - Determines if a rule exists whose port range ends at the specified number. The word 'to_' does *not* relate to inbound/outbound directionality; it relates to the port range ("counting _to_"). `to_port` is an exact criterion; so if the rule allows 1000-2000 and you specify a `to_port` of 1999, it does not match.
-
-    describe aws_security_group(group_name: 'mixed-functionality-group') do
-      # Allow RDP from defined range
-      it { should allow_in(port: 3389, ipv4_range: '10.5.0.0/16') }
-
-      # Allow SSH from two ranges
-      it { should allow_in(port: 22, ipv4_range: ['10.5.0.0/16', '10.2.3.0/24']) }
-
-      # Check Bacula port range
-      it { should allow_in(from_port: 9101, to_port: 9103, ipv4_range: '10.6.7.0/24') }
-
-      # Assuming the AWS SG allows 9001-9003, use inexact matching to check 9002
-      it { should allow_in(port: 9002) }
-
-      # Assuming the AWS SG allows 10.2.1.0/24, use inexact matching to check 10.2.1.33/32
-      it { should allow_in(ipv4_range: '10.2.1.33/32') }
-
-      # Ensure the 3rd outbound rule is TCP-based
-      it { should allow_in(protocol: 'tcp', position: 3') }
-
-      # Do not allow unrestricted IPv4 access.
-      it { should_not allow_in(ipv4_range: '0.0.0.0/0') }
+  - from\_port - Determines if a rule exists whose port range begins at the specified number. The word 'from\_' does *not* relate to inbound/outbound directionality; it relates to the port range ("counting *from*"). `from_port` is an exact criterion; so if the rule allows 1000-2000 and you specify a `from_port` of 1001, it does not match.
+
+  - ipv4\_range - Specifies an IPv4 address or subnet as a CIDR, or a list of them, to be checked as a permissible origin (for `allow_in`) or destination (for `allow_out`) for traffic.  Each AWS Security Group rule may have multiple allowed source IP ranges.
+
+  - port - Determines if a particular TCP/IP port is reachable. allow\_in and allow\_out examine whether the specified port is included in the port range of a rule, while allow\_in. You may specify the port as a string (`'22'`) or as a number.
+
+  - position - A one-based index into the list of rules. If provided, this restricts the evaluation to the rule at that position. You may also use the special values `:first` and `:last`. `position` may also be used to enable `allow_in_only` and `allow_out_only` to work with multi-rule Security Groups.
+
+  - protocol - Specifies the IP protocol. 'tcp', 'udp', and 'icmp' are some typical values. The string "-1" or 'any' is used to indicate any protocol.
+
+  - to\_port - Determines if a rule exists whose port range ends at the specified number. The word 'to\_' does *not* relate to inbound/outbound directionality; it relates to the port range ("counting *to*"). `to_port` is an exact criterion; so if the rule allows 1000-2000 and you specify a `to_port` of 1999, it does not match.
+    
+    describe aws\_security\_group(group\_name: 'mixed-functionality-group') do
```

Comment:
I think the lines below need to be cleaned up by using backticks and not escaping the underscores.